src/FakeItEasy/Expressions/CallExpressionParser.cs

```diff
@@ -39,7 +32,7 @@ public ParsedCallExpression Parse(LambdaExpression callExpression, object fake)
         private static ParsedCallExpression ParseInvocationExpression(InvocationExpression expression)
         {
             var target = expression.Expression.Evaluate();
-            var method = target.GetType().GetMethod("Invoke");
+            var method = target!.GetType().GetMethod("Invoke");

```

Comment:
Can we be sure `target` will never be null? It seems a pretty big assumption, when invoking an expression we know nothing about... Not sure what to do if it's null, though... I guess we'll just throw `NullReferenceException`. Which makes sense, since it would mean the user did something like this:
```csharp
Action action = null;
A.CallTo(() => action())...
```

Reply:
Anyway, do we really need to evaluate the expression here? We only use `target` to get its type, but we could get the type from `expression.Expression.Type`. Just tried it, it seems to work.